src/lib/frontend/d_cnf.ml

```diff
@@ -729,7 +728,7 @@ let mk_mr_ty_decls (tdl: DE.ty_cst list) =
             ) cases ([], true)
           in
           let uid = Uid.of_dolmen ty_c in
-          if is_enum
+          if is_enum && not contains_adts
```

Comment:
Before this PR, we sent enum types in mutually recursive definitions to the `Sum` theory. We cannot do that anymore because we need to send wellf-founded nest to the ADT theory. I didn't write a comment here because this code will disappear after merging `Sum` theory into `ADT` theory.

Reply:
I think it's okay to remove it so I do.